### ISSUE TITLE
请问，复现成功了吗

### ISSUE THREAD

**Issue by 605436079** [guruL/Character-Region-Awareness-for-Text-Detection-#1]
最近看到这篇论文，感觉准确率应该还可以，就是单字符级标签比较难弄。请问大神复现结果怎么样

**Comment by LoveGalaxy** [comment]
正在复现中，可以一起交流，字符级主要是靠生成的样本集

**Comment by SakuraRiven** [comment]
Hi，请问复现的效果怎么样呢，有没有达到文章里面的 95 Fscore，我的复现目前只有91多

**Comment by LoveGalaxy** [comment]
> Hi，请问复现的效果怎么样呢，有没有达到文章里面的 95 Fscore，我的复现目前只有91多
最近比较忙，还没有复现成功，请问你会开源吗

**Comment by 605436079** [comment]
作者又不是我！！！我咋开源。是一家韩国公司发的论文，开源可能性应该比较小------------------ 原始邮件 ------------------
发件人: "DayDayUp"<notifications@github.com>
发送时间: 2019年6月11日(星期二) 中午1:49
收件人: "guruL/Character-Region-Awareness-for-Text-Detection-"<Character-Region-Awareness-for-Text-Detection-@noreply.github.com>;
抄送: "605436079"<605436079@qq.com>;"Author"<author@noreply.github.com>;
主题: Re: [guruL/Character-Region-Awareness-for-Text-Detection-] 请问，复现成功了吗 (#1)


 
Hi，请问复现的效果怎么样呢，有没有达到文章里面的 95 Fscore，我的复现目前只有91多
 最近比较忙，还没有复现成功，请问你会开源吗
  
&mdash;

**Comment by SakuraRiven** [comment]
@guruL 因为某些原因，暂时没办法开源，有问题可以一起讨论哈，希望可以复现出论文的精度

**Comment by LoveGalaxy** [comment]
好的，主要有三个问题想请教一下，在论文中好像没有写得很仔细，请教下你在复现的时候是怎么做的呢？
1. 请问在生成数据集上用的什么loss呢？
2. 做过哪些数据增强吗？
3. 两个gt的高斯分布的细节？比如sigma大小会随着box大小变化吗？还有边缘值会保证有一个最小的值吗？类似这样的细节？

**Comment by SakuraRiven** [comment]
1. 仿真数据上直接用L2 Loss就可以了，pretrain的效果挺不错的，在13上可以直接达到快80的Fscore
2. finetune的时候有用常见的resize、rotate和crop吧，感觉好像对结果影响不大
3. 按论文里说的，事先准备高斯模板然后直接warp到目标上就行，sigma这得肉眼观察下，差不多就可以了

**Comment by LoveGalaxy** [comment]
IC13有两个Task，对应两种标注方式，请问你finetune的时候用的 word 级来半监督，还是直接用的 character 级的标注呢
我训练仿真数据的时候有一个问题，容易对背景产生一些细微的响应，论文的图示给我的感觉是对背景的响应很干净。我想请问下你有遇到这个问题吗？响应的位置有些是和文字有类似纹理的地方，有些也不是，直接CCL会生成一些错误的小框，是使用一些阈值过滤了吗？

**Comment by SakuraRiven** [comment]
paper里拿13和17一起训的，所以我13也统一用的半监督，可能用char级好一点？不确定

在最后finetune出结果的时候有过滤，pretrain感觉不需要

**Comment by LoveGalaxy** [comment]
预训练了多少个epoch啊
感觉可以混合学习吧，有字符集肯定更准确吧，他在fine-tune的时候不是还加了合成数据吗？
我现在预训练的最大问题就是错误响应有点多，大尺度的字符定位也有点问题

**Comment by SakuraRiven** [comment]
和论文一样50k，不过25k的时候就已经很不错了
finetune的时候是有1/6概率加入合成数据的，13的话使用char标注应该会好点吧，我之后会尝试一下

**Comment by LoveGalaxy** [comment]
不好意思啊，再请教个问题，请问 OHNM 你是怎么做的呢？直接在 MSE Loss 上做吗

**Comment by SakuraRiven** [comment]
这个我也不确定，就只是取MSE Loss的负样本topk和正样本来bp的

**Comment by LoveGalaxy** [comment]
你好啊，我现在预训练在IC13上的结果要比以前好些了，但是我观察在合成数据上其实效果还不好，我有几个地方想再请教一下你的设置，一个是batch_size 一个是训练的图片size

**Comment by SakuraRiven** [comment]
论文里面是crop的640，这个按照他的就好了啊，我的话在800k上预训练1个epoch就可以了

**Comment by LoveGalaxy** [comment]
@SakuraRiven 请问你还在复现吗，结果有多少了呢
你可以看下这个 https://github.com/clovaai/CRAFT-pytorch
我用它预训练的模型，在他的测试代码里测了下 93左右 自己写的代码测了下只有80多，说明解码应该影响也蛮大的，可以参考下它解码代码。

**Comment by SakuraRiven** [comment]
之前调参调到92点多，后来有其他事情所以就不再关注这个方法了。
我看官方没有开源训练代码，所以说实话能参考的有限。我没有测他的代码，不过你说的93左右，是13+17finetune的模型吗？你自己的80多也是指13+17finetune的模型吗

**Comment by LoveGalaxy** [comment]
是的，都是13+17 finetune 但还是比论文的还差两个点，他的源码会输出4个点，IC13是测试2个点，我改了下直接输出了最大最小的值

**Comment by SakuraRiven** [comment]
我这边改4个点或者2个点对结果影响不大，主要对finetune细节挺好奇的，但可惜没放出来

我在想我把13换成char级别标注，可能会再高一点，诸如此类还有很多可以调的地方，不过感觉没啥意思了。后面因为有其他的事，所以就不关注了。加油，希望你可以早一点复现出论文精度哈~

**Comment by namedysx** [comment]
请问复现了吗

**Comment by wulitaotao1** [comment]
请问在预训练过程中，你有碰到召回率急剧下降的情况吗？

**Comment by mmmmmore** [comment]
@SakuraRiven @guruL 请问有什么调参技巧吗？在SynthText上训练到了F1多少才去finetune ?

**Comment by BaiZhiqi** [comment]
@SakuraRiven 你好，请问你现在可以开源你的训练过程了嘛？我也在复现该模型，遇到了一些问题
